gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/api/use_case/MigrateApiUseCaseTest.java

```diff
@@ -256,6 +268,182 @@ void should_return_fail_when_api_dont_use_v4_emulation_engine() {
+        assertThat(result.state()).isEqualTo(MigrationResult.State.MIGRATED);
+    }
+
+    @Test
+    void should_migrate_when_api_has_ASCIIDOC_page() {
+        // Given
+        var api = ApiFixtures.aProxyApiV2().toBuilder().id(API_ID).build();
+        api.getApiDefinition().setExecutionMode(ExecutionMode.V4_EMULATION_ENGINE);
+        PageEntity publishedPage = new PageEntity();
+        publishedPage.setPublished(true);
+        PageEntity asciidocTemplatePage = new PageEntity();
+        asciidocTemplatePage.setPublished(true);
+        asciidocTemplatePage.setType(PageType.ASCIIDOC.name());
+        doReturn(Arrays.asList(publishedPage, asciidocTemplatePage)).when(pageService).findByApi(any(), any());
+        apiCrudService.initWith(List.of(api));
+
+        // When
+        var result = useCase.execute(new MigrateApiUseCase.Input(API_ID, null, AUDIT_INFO));
+
+        // Then
+        assertThat(result.apiId()).isEqualTo(API_ID);
+        assertThat(result.state()).isEqualTo(MigrationResult.State.MIGRATED);
+    }
+
+    @Test
+    void should_migrate_when_api_has_OPENAPI_page() {
+        // Given
+        var api = ApiFixtures.aProxyApiV2().toBuilder().id(API_ID).build();
+        api.getApiDefinition().setExecutionMode(ExecutionMode.V4_EMULATION_ENGINE);
+        PageEntity publishedPage = new PageEntity();
+        publishedPage.setPublished(true);
+        PageEntity swaggerTemplatePage = new PageEntity();
+        swaggerTemplatePage.setPublished(true);
+        swaggerTemplatePage.setType(PageType.SWAGGER.name());
+        doReturn(Arrays.asList(publishedPage, swaggerTemplatePage)).when(pageService).findByApi(any(), any());
+        apiCrudService.initWith(List.of(api));
+
+        // When
+        var result = useCase.execute(new MigrateApiUseCase.Input(API_ID, null, AUDIT_INFO));
+
+        // Then
+        assertThat(result.apiId()).isEqualTo(API_ID);
+        assertThat(result.state()).isEqualTo(MigrationResult.State.MIGRATED);
+    }
+
+    @Test
+    void should_migrate_when_api_has_ASYNCAPI_page() {
+        // Given
+        var api = ApiFixtures.aProxyApiV2().toBuilder().id(API_ID).build();
+        api.getApiDefinition().setExecutionMode(ExecutionMode.V4_EMULATION_ENGINE);
+        PageEntity publishedPage = new PageEntity();
+        publishedPage.setPublished(true);
+        PageEntity asyncapiTemplatePage = new PageEntity();
+        asyncapiTemplatePage.setPublished(true);
+        asyncapiTemplatePage.setType(PageType.ASYNCAPI.name());
+        doReturn(Arrays.asList(publishedPage, asyncapiTemplatePage)).when(pageService).findByApi(any(), any());
+        apiCrudService.initWith(List.of(api));
+
+        // When
+        var result = useCase.execute(new MigrateApiUseCase.Input(API_ID, null, AUDIT_INFO));
+
+        // Then
+        assertThat(result.apiId()).isEqualTo(API_ID);
+        assertThat(result.state()).isEqualTo(MigrationResult.State.MIGRATED);
+    }
```

Comment:
the OK tests can be replaced by parametrized tests


```suggestion
    @ParametrizedTest
    @EnumSource(PageType.class)
    void should_migrate_when_api_has_documentation_page(PageType type) {
        // Given
        var api = ApiFixtures.aProxyApiV2().toBuilder().id(API_ID).build();
        api.getApiDefinition().setExecutionMode(ExecutionMode.V4_EMULATION_ENGINE);
        PageEntity publishedPage = new PageEntity();
        publishedPage.setPublished(true);
        PageEntity page = new PageEntity();
        page.setPublished(true);
        page.setType(type.name());
        when(pageService.findByApi(any(), any())).thenReturn(List.of(publishedPage, page));
        apiCrudService.initWith(List.of(api));

        // When
        var result = useCase.execute(new MigrateApiUseCase.Input(API_ID, null, AUDIT_INFO));

        // Then
        assertThat(result.apiId()).isEqualTo(API_ID);
        assertThat(result.state()).isEqualTo(MigrationResult.State.MIGRATED);
    }
```

Reply:
fixed